python/ray/_private/services.py

```diff
@@ -919,7 +919,39 @@ def start_redis(node_ip_address,
         # If no port is given, fallback to default Redis port for the primary
         # shard.
         if port is None:
-            port = ray_constants.DEFAULT_PORT
+            default_port = ray_constants.DEFAULT_PORT
+            if sys.platform == "win32":
+                # In Windows, the port for Redis server is selected by first
+                # trying on to obtain the default_port i.e., 6379. If it's
+                # not available or if it's not possible to use it then we
+                # randomly try to select other ports for starting Redis
+                # servers for the current Ray instance. The reason for
+                # random selection of ports is to lower the chances of two
+                # ray instances attempting to acquire the same port at some
+                # point of time. This helps in reducing the time spent in
+                # the following for loop.
+                port = default_port  # Start from default_port
+                port_found = False
+                num_attempts = 200  # The number of attempts to try other ports
+                a_socket = socket.socket(socket.AF_INET,
+                                         socket.SOCK_STREAM)  # A TCP socket
+                for _ in range(num_attempts):
+                    location = (node_ip_address, port)
+                    try:
+                        a_socket.bind(location)
+                        port_found = True
+                        break
+                    except socket.error:
+                        # location is not available for use
+                        # Select next port to be tried randomly
+                        port += random.randint(-100, 100)
```

Comment:
It would be slightly preferable to do `port = default_port + random.randint(-100, 100)`, otherwise we are doing a random walk in port space here which is a bit less desirable (but no big deal).

We need to re-run the test anyways, so I'll just push that change to your branch :)